komga-webui/src/functions/reader.ts

```diff
@@ -0,0 +1,396 @@
+import KomgaBooksService from '@/services/komga-books.service'
+import { BackgroundColors, ImageFit, ReadingDirection } from '@/types/enum-books'
+import { Map } from '@/functions/forms'
+import KomgaSeriesService from '@/services/komga-series.service'
+import { getBookTitleCompact } from '@/functions/book-title'
+import { Cookie, Mapper } from '@/functions/cookies'
+import { VueCookies } from 'vue-cookies'
+import { bookPageUrl } from '@/functions/urls'
+import { checkWebpFeature } from '@/functions/check-webp'
+import {
+  DoubleSpreadBuilder,
+  PageSpread,
+  PageSpreadBuilder, PageSpreads,
+  SingleSpreadBuilder,
+  WebToonBuilder,
+} from '@/functions/pages'
+import { VueRouter } from 'vue-router/types/router'
+
+export type Optional<V> = V | undefined
```

Comment:
not sure how much this brings compared to `V | undefined`. This could anyhow be moved to a more generic location if we really need it.